libs/api_common/device_helper.cpp

```diff
@@ -59,3 +68,147 @@ get_device_handle()
 
     return _device_handle;
 }
+
+void
+cleanup_async_ioctl_completion(_Inout_opt_ _Post_invalid_ async_ioctl_completion_t* async_ioctl_completion)
+{
+    if (async_ioctl_completion != nullptr) {
+        if (async_ioctl_completion->wait != nullptr) {
+            // Unregister the wait by setting the event to NULL.
+            SetThreadpoolWait(async_ioctl_completion->wait, nullptr, nullptr);
+
+            // Close the wait.
+            CloseThreadpoolWait(async_ioctl_completion->wait);
+        }
+
+        if (async_ioctl_completion->overlapped.hEvent != nullptr)
+            ::CloseHandle(async_ioctl_completion->overlapped.hEvent);
+
+        ebpf_free(async_ioctl_completion);
+    }
+}
+
+ebpf_result_t
+register_wait_async_ioctl_operation(_Inout_ async_ioctl_completion_t* async_ioctl_completion)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+
+    EBPF_LOG_ENTRY();
+
+    // This function registers wait for the async IOCTL operation completion, by resetting
+    // the OVERLAPPED structure used for the async IOCTL, resetting the event object used in the previous call,
+    // and arming the threadpool object once more with the event handle.
+
+    // Save the overlapped event handle.
+    HANDLE event = async_ioctl_completion->overlapped.hEvent;
+
+    // Reset the overlapped object.
+    memset(&async_ioctl_completion->overlapped, 0, sizeof(OVERLAPPED));
+
+    if (event == nullptr) {
+        // Create a new event object for OVERLAPPED struct.
+        async_ioctl_completion->overlapped.hEvent = CreateEvent(nullptr, true /* manual reset */, false, nullptr);
```

Comment:
For comparison, svcmain.cpp comments all the parameters like so:
```
    ebpf_service_stop_event_handle = CreateEvent(
        nullptr,  // default security attributes
        true,     // manual reset event
        false,    // not signaled
        nullptr); // no name
```
Suggest doing the same here.

```suggestion
        async_ioctl_completion->overlapped.hEvent = CreateEvent(
            nullptr,  // default security attributes
            true,     // manual reset event
            false,    // not signaled
            nullptr); // no name
```